hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -432,6 +433,11 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() {
               table.getActiveTimeline(), config.getInlineCompactDeltaCommitMax())
               : Option.empty();
 
+      // The clustering commit instant can not be archived unless we ensure that the replaced files have been cleaned,
+      // without the replaced files metadata on the timeline, the fs view would expose duplicates for readers.
+      Option<HoodieInstant> oldestInstantToRetainForClustering =
```

Comment:
this name is a little confused with `oldestPendingCompactionAndReplaceInstant` below.

Reply:
@leesf, refer to the naming of `oldestInstantToRetainForCompaction`.